scripts/update-extension.sh

```diff
@@ -4,27 +4,34 @@ source .env
 
 RELEASE_NAMESPACE="${RELEASE_NAMESPACE:-arc-osm-system}"
 EXTENSION_NAME="${EXTENSION_NAME:-osm}"
-EXTENSION_TYPE="${EXTENSION_TYPE:-Microsoft.openservicemesh}"
 EXTENSION_SETTINGS=$1
-API_VERSION="${API_VERSION:-2020-07-01-preview}"
 
-export RESOURCEID=subscriptions/$SUBSCRIPTION/resourceGroups/$RESOURCEGROUP/providers/Microsoft.Kubernetes/connectedClusters/$CLUSTERNAME
+az account set --subscription="$SUBSCRIPTION" > /dev/null 2>&1
 
-echo "Azure Resource ID: $RESOURCEID"
+# confirm
+helm ls --all --all-namespaces
 
 if [[ -z "$EXTENSION_SETTINGS" ]]; then
-    EXTENSION_SETTINGS="osm_extension.json"
-    jq -n \
-        --arg tag "$EXTENSION_TAG" \
-        --arg namespace "$RELEASE_NAMESPACE" \
-        '{properties: {extensionType: "Microsoft.openservicemesh", autoUpgradeMinorVersion: "false", version: $tag, releaseTrain: "Staging", scope: { cluster: { releaseNamespace: $namespace } }, "configurationProtectedSettings": { "osm.OpenServiceMesh.controllerLogLevel": "debug", } } }' > osm_extension.json
+   az k8s-extension create \
+      --cluster-name $CLUSTERNAME \
+      --resource-group $RESOURCEGROUP \
+      --cluster-type connectedClusters \
+      --extension-type Microsoft.openservicemesh \
+      --scope cluster \
+      --release-train staging \
+      --name $EXTENSION_NAME \
+      --release-namespace $RELEASE_NAMESPACE \
+      --version $EXTENSION_TAG
+else
+   az k8s-extension create \
+      --cluster-name $CLUSTERNAME \
+      --resource-group $RESOURCEGROUP \
+      --cluster-type connectedClusters \
+      --extension-type Microsoft.openservicemesh \
+      --scope cluster \
+      --release-train staging \
+      --name $EXTENSION_NAME \
+      --release-namespace $RELEASE_NAMESPACE \
+      --version $EXTENSION_TAG \
+      --configuration-settings-file $EXTENSION_SETTINGS
```

Comment:
is there a way to do this without the code replication? like set all the args once as a string or something, optionally add config settings and then call az k8s-extension create? not a big deal, just may be a bit cleaner

Reply:
Will look into this in a follow up PR if I think of something.